pyat/at/lattice/variable_elements.py

```diff
@@ -68,74 +89,103 @@ def __init__(self, family_name, AmplitudeA=None, AmplitudeB=None,
 
         .. note::
 
-            * For all excitation modes all least one amplitude has to be provided.
+            * If no parameters are given it will be initialized as IdentityPass.
+            * For all excitation modes at least one amplitudes (A or B) has
+              to be provided.
               The default excitation is ``ACMode.SINE``
             * For ``mode=ACMode.SINE`` the ``Frequency(A,B)`` corresponding to the
-              ``Amplitude(A,B)`` has to be provided
+              ``Amplitude(A,B)`` has to be provided.
             * For ``mode=ACMode.ARBITRARY`` the ``Func(A,B)`` corresponding to the
-              ``Amplitude(A,B)`` has to be provided
+              ``Amplitude(A,B)`` has to be provided.
         """
-        kwargs.setdefault('PassMethod', 'VariableThinMPolePass')
-        self.MaxOrder = kwargs.pop('MaxOrder', 0)
-        self.Periodic = kwargs.pop('Periodic', True)
-        self.Mode = int(mode)
-        if AmplitudeA is None and AmplitudeB is None:
-            raise AtError('Please provide at least one amplitude for A or B')
-        AmplitudeB = self._set_params(AmplitudeB, mode, 'B', **kwargs)
-        AmplitudeA = self._set_params(AmplitudeA, mode, 'A', **kwargs)
-        self._setmaxorder(AmplitudeA, AmplitudeB)
-        if mode == ACMode.WHITENOISE:
-            self.Seed = kwargs.pop('Seed', datetime.now().timestamp())
-        self.PolynomA = numpy.zeros(self.MaxOrder+1)
-        self.PolynomB = numpy.zeros(self.MaxOrder+1)
-        ramps = kwargs.pop('Ramps', None)
-        if ramps is not None:
-            assert len(ramps)==4, \
-                'Ramps has to be a vector with 4 elements'
-            self.Ramps = ramps
-        super(VariableMultipole, self).__init__(family_name, **kwargs)
-
-    def _setmaxorder(self, ampa, ampb):
+        if len(kwargs) > 0:
```

Comment:
This was needed in order to separate the case when the element is called inside pyat, therefore the object already exists and there are no kwargs, from the case when the element is instantiated using a dictionary (the dictionary comming from reading a .mat, or from user settings).

Sometimes the object is read, for example when printing in a terminal, the element is passed again (I guess this happens because it inherits `__str__` from other element) and no modifications in the object are needed, I just need to pass it to super and find its way to `__str__`.